File: scala/grade-school/src/main/scala/grade_school.scala
import scala.collection.immutable.SortedMap

class School {

  private var students = SortedMap[Int, Seq[String]]().withDefaultValue(Vector[String]())

  def db: Map[Int, Seq[String]] = students

  def add(name: String, grade: Int): Unit = {    
    students += (grade -> (students(grade) :+ name))
  }

  def grade(grade: Int): Seq[String] = {
    students(grade)
  }

  def sorted(): Map[Int, Seq[String]] = {
    students.mapValues(_.sorted)
  } 

}

File: python/grade-school/grade_school.py
from collections import defaultdict, OrderedDict

class School(object):
  def __init__(self, name):
    self._name = name
    self._roster = defaultdict(list)

  def add(self, student, grade):
    self._roster[grade].append(student)

  def grade(self, num):
    return list(self._roster[num])

  def sort(self):
    return OrderedDict((grade, tuple(sorted(students))) for grade, students in sorted(self._roster.items()))
